src/lndk_offers.rs

```diff
@@ -1,7 +1,189 @@
+use crate::blocking_client::BlockingClient;
+use bitcoin::network::constants::Network;
+use bitcoin::secp256k1::schnorr::Signature;
+use bitcoin::secp256k1::{Error as Secp256k1Error, Message, PublicKey};
+use lightning::offers::invoice_request::InvoiceRequest;
 use lightning::offers::offer::Offer;
 use lightning::offers::parse::Bolt12ParseError;
+use tonic_lnd::signrpc::{KeyLocator, SignMessageReq};
+use tonic_lnd::tonic::Status;
 
 // Decodes a bech32 string into an LDK offer.
 pub fn decode(offer_str: String) -> Result<Offer, Bolt12ParseError> {
     offer_str.parse::<Offer>()
 }
+
+#[allow(dead_code)]
+pub(crate) fn request_invoice(
+    mut signer: impl MessageSigner,
+    offer: Offer,
+    metadata: Vec<u8>,
+    network: Network,
+    msats: u64,
+) -> InvoiceRequest {
```

Comment:
Return a `Result<InvoiceRequest, OfferError>` rather than using `expect` and `panic` here?